q2_feature_table/_summarize/summarize_assets/src/chart.js

```diff
@@ -0,0 +1,154 @@
+import * as d3 from 'd3';
+
+
+const categoricalBin = (data, counts, x) => {
+  const result = [];
+  x.domain().forEach((cat) => {
+    const catArray = Object.keys(data)
+                      .filter(sampleID => data[sampleID] === cat)
+                      .map(sampleID => counts[sampleID]);
+    catArray.x0 = x(cat);
+    result.push(catArray);
+  });
+  return result;
+};
+
+
+const updateChart = (bar, props, y, threshold) => {
+  bar.selectAll('.overlay')
+    .attr('y', d => y(d.filter(count => count >= threshold).length))
+    .attr('height', d => props.height - y(d.filter(count => count >= threshold).length));
+};
+
+const buildChart = (svg, props, metadata, counts) => {
+  svg.selectAll('*').remove();
+
+  const threshold = d3.select('#slider').node().min;
+  const option = d3.select('select').node().value;
+  const data = metadata[option];
+  const categories = [...new Set(Object.keys(data).map(key => data[key]))];
+
+  categories.sort(naturalSort);
+
+  const g = svg
+      .append('g')
+    .attr('transform', `translate(${props.margin.left}, ${props.margin.top})`);
+
+  const x = d3
+    .scaleBand()
+    .domain(categories)
+    .range([0, props.width])
+    .paddingInner([0.1])
+    .paddingOuter([0.3]);
+
+  const bins = categoricalBin(data, counts, x);
+
+  const y = d3
+    .scaleLinear()
+    .domain([0, d3.max(bins, d => d.length)])
+    .range([props.height, 0]);
+
+  g.append('g')
+    .attr('class', 'axis axis--x')
+    .attr('transform', `translate(0, ${props.height})`)
+    .call(d3.axisBottom(x));
+
+  // vertical axis labels if there are a good amount of categories
+  if (categories.length > 5) {
+    g.selectAll('text')
+      .attr('y', 0)
+      .attr('x', 9)
+      .attr('dy', '.35em')
+      .attr('transform', 'rotate(90)')
+      .style('text-anchor', 'start');
+  }
+
+  g.append('g')
+    .attr('class', 'axis axis--y')
+    .call(d3.axisLeft(y));
+
+  const bar = g
+      .selectAll('.bar')
+    .data(bins)
+      .enter()
+    .append('g')
+    .attr('class', 'bar');
+
+  // build backdrop
+  bar
+      .append('rect')
+    .attr('class', 'original')
+    .attr('fill', 'black')
+    .attr('opacity', 0.15)
+    .attr('x', d => d.x0)
+    .attr('y', d => y(d.length))
+    .attr('width', x.bandwidth() - 1)
+    .attr('height', d => props.height - y(d.length));
+
+  // build overlay
+  bar
+      .append('g')
+    .attr('class', 'overlay-group')
+      .append('rect')
+    .attr('class', 'overlay')
+    .attr('fill', 'steelblue')
+    .style('opacity', 1)
+    .style('fill-opacity', 1)
+    .attr('stroke', 'lightgray')
+    .attr('x', d => d.x0)
+    .attr('y', d => y(d.filter(count => count >= threshold).length))
+    .attr('width', x.bandwidth() - 1)
+    .attr('height', d => props.height - y(d.filter(count => count >= threshold).length));
+
+  const hiddenDepths = d3.select('#hidden-depths');
+  hiddenDepths
+    .on('change', () => updateChart(bar, props, y, +hiddenDepths.node().value));
```

Comment:
I think we need to updateChart at the end, so that if the category changes (causing a rebuild) the existing depth is used (it hasn't changed, so the on-change handler won't fire)